packages/core/src/services/worldstate/context-builder.ts

```
@@ -17,11 +29,125 @@ export class ContextBuilder {
+        switch (stimulus.type) {
+            case StimulusSource.UserMessage:
+                return this.buildFromUserMessage(stimulus as UserMessageStimulus);
+            case StimulusSource.SystemEvent:
+                return this.buildFromSystemEvent(stimulus as SystemEventStimulus);
+            case StimulusSource.ScheduledTask:
+                return this.buildFromScheduledTask(stimulus as ScheduledTaskStimulus);
+            case StimulusSource.BackgroundTaskCompletion:
+                return this.buildFromBackgroundTask(stimulus as BackgroundTaskCompletionStimulus);
+            default:
+                const _exhaustive: never = stimulus;
+                throw new Error(`Unsupported stimulus type: ${(stimulus as any).type}`);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)